pkg/cloud-init/cloud-init.go

```diff
@@ -206,7 +206,7 @@ func GenerateLocalData(vmName string, namespace string, source *v1.CloudInitNoCl
 	} else {
 		return errors.New(fmt.Sprintf("userDataBase64 or userData is required for no-cloud data source"))
 	}
-	metaData := []byte(fmt.Sprintf("{ \"instance-id\": \"%s.%s\" }\n", vmName, namespace))
+	metaData := []byte(fmt.Sprintf("{ \"instance-id\": \"%s.vms.%s\", \"local-hostname\": \"%s.vms.%s\" }\n", vmName, namespace, vmName, namespace))
```

Comment:
Hm, we definitely should not add the namespace to the hostname. A typical resolv.conf for a pod in the default  namespace looks like this:

```
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local
options ndots:5
```

A host called `myvm.vms.default` would search for `myvm.vms.default.default...` and therefore can't resolve its own fqdn.

However, if we go with `subdomain=vms` on the pods, it might make sense to name the vm `myvm.vms`, since k8s does not add the subdomain search path (see https://github.com/kubernetes/kubernetes/issues/48019 for details).

We would not need the subdomain to integrate the vm with kube-dns, I am just a little bit worried, that it will be confusing for people, that having a vm called `test` and a pod called `test` would conflict, since they are on different levels. But maybe we should **not** use the subdomain for now. @karmab, @vladikr thoughts?

A final option is to add the subdomain searchpath ourselves when we offer them via dhcp, however then we might run into glibc issues on the vms themselves when they use old glibc versions.